### PR TITLE
Caliper v1p2 update

### PR DESCRIPTION
**Pull request by andrew-gardener**

### CODE REVIEW

**Comment by andrew-gardener** [comment]
Need to wait on Caliper 1.2 php implementation library release

File: app/libs/caliper/entity.php [hunk 1]
```diff
@@ -60,9 +57,44 @@ public static function session() {
...
+		if ( array_key_exists( 'HTTP_HOST', $_SERVER ) ) {
+			$user_client->setHost( $_SERVER['HTTP_HOST'] );
+		}
+
+		if ( array_key_exists( 'HTTP_USER_AGENT', $_SERVER ) ) {
+			$user_client->setUserAgent( $_SERVER['HTTP_USER_AGENT'] );
+		}
+
+		// get ip address if available.
+		if ( ! empty( $_SERVER['HTTP_CLIENT_IP'] ) ) {
```
Comment (kitsook) [inline app/libs/caliper/entity.php:87]: Should we take the priority as `HTTP_X_FORWARDED_FOR`, then `REMOTE_ADDR`, and put `HTTP_CLIENT_IP` as the last resort?  Users can spoof the  `HTTP_CLIENT_IP` value by sending http header `CLIENT-IP`.

But then, if the server is not behind proxy, users can fool us by setting `X-FORWARDED-FOR` header. (If the server is behind proxy, hopefully the proxy will chain up `HTTP_X_FORWARDED_FOR` correctly). `REMOTE_ADDR` is the best if server is not behind proxy as it is set by the server.  Maybe we should log both `HTTP_X_FORWARDED_FOR` and `REMOTE_ADDR`?

Comment (kitsook) [reply app/libs/caliper/entity.php:87]: interesting read on CakePHP's `clientIp` function vulnerabilities

https://legalhackers.com/advisories/CakePHP-IP-Spoofing-Vulnerability.html

Comment (andrew-gardener) [reply app/libs/caliper/entity.php:87]: Really good point, I'll rearrange them. I didn't put to much through into the order originally (I think I originally just plopped in a list I found of most common possible variables ip addresses can appear in).

File: app/libs/caliper/entity.php [hunk 2]
```diff
@@ -48,7 +44,8 @@ public static function session() {
         $session_end = $user_session->read('session_end');
 
         $session = (new Session( ResourceIRI::user_session($session_id)) )
-            ->setUser(CaliperActor::generateActor($user));
+            ->setUser(CaliperActor::generateActor($user))
+            ->setClient( CaliperEntity::client( $session_id ) );
```
[the remark this answers is not in the document]
Comment (andrew-gardener) [reply app/libs/caliper/entity.php:48]: We could in theory use something else like creating a secondary id in the session just for caliper use (the more important part is that the id is attached to the same session). 
The endpoint that the information is sent to should be a highly trusted though (and events should always be sent through https). 

@xcompass what's your opinion on this?

[the remark this answers is not in the document]
Comment (andrew-gardener) [reply app/libs/caliper/entity.php:48]: I'm going to go ahead assuming that its alright for now and I can change it later if there is an issue.

[the remark this answers is not in the document]
Comment (xcompass) [reply app/libs/caliper/entity.php:48]: How about sending a hash of session_id?

[the remark this answers is not in the document]
Comment (andrew-gardener) [reply app/libs/caliper/entity.php:48]: that sounds good, will add

**Pull request merged**